### ISSUE TITLE
Add trendy models/gen methods

### ISSUE THREAD

**Issue by one-some** [one-some/KoboldAI-united#7]
thanks to not being restrained by hf nonsense we can probably begin implementing trendy stuff hf doesnt have. for a backend to have full integration in kobold, it needs to support callbacks for viewing and manipulating token logits as they're generated.

some stuff to look into:
- [ ] ggml/llama.cpp -- Really fast cpu inference. Written in C++, python binding available [here](https://github.com/abetlen/llama-cpp-python). not sure if it supports all we need it to but its worth a looksie
- [ ] rwkv -- RNN based language model, module interface available [here](https://github.com/BlinkDL/ChatRWKV/tree/main/rwkv_pip_package). kind of implemented in kobold but implementation is rough
